browser/search_engines/search_engine_tracker_browsertest.cc

```diff
@@ -26,14 +27,28 @@ class SearchEngineProviderP3ATest : public InProcessBrowserTest {
   std::unique_ptr<base::HistogramTester> histogram_tester_;
 };
 
-IN_PROC_BROWSER_TEST_F(SearchEngineProviderP3ATest,
-                       DISABLED_DefaultSearchEngineP3A) {
+testing::AssertionResult VerifyTemplateURLServiceLoad(
+    TemplateURLService* service) {
+  if (service->loaded()) {
+    return testing::AssertionSuccess();
+  }
+  search_test_utils::WaitForTemplateURLServiceToLoad(service);
+  if (service->loaded()) {
+    return testing::AssertionSuccess();
+  }
+  return testing::AssertionFailure() << "TemplateURLService isn't loaded";
+}
+
+IN_PROC_BROWSER_TEST_F(SearchEngineProviderP3ATest, DefaultSearchEngineP3A) {
   // Check that the metric is reported on startup.
   histogram_tester_->ExpectUniqueSample(kDefaultSearchEngineMetric,
                                         SearchEngineP3A::kGoogle, 1);
 
-  auto* service = TemplateURLServiceFactory::GetForProfile(
-      browser()->profile());
+  auto* service =
+      TemplateURLServiceFactory::GetForProfile(browser()->profile());
+
+  // Make sure the service is initialized.
+  EXPECT_TRUE(VerifyTemplateURLServiceLoad(service));
```

Comment:
frankly speaking I don't quite understand why simple `search_test_utils::WaitForTemplateURLServiceToLoad(service);` is not enough. It already asserts that the service is loaded inside, so this wrapper looks really redundant

Reply:
Seems fine to me too! I've made it so.